source/common/router/router.cc

```diff
@@ -2096,6 +2102,38 @@ bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
   return false;
 }
 
+void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or_trailers,
+                                        UpstreamRequest& upstream_request) {
+  // Process the load report only once, so if response has report in headers,
+  // then don't process it in trailers.
+  if (orca_load_report_received_) {
+    return;
+  }
+  // Check whether we need to send the load report to the LRS or invoke the ORCA
+  // callbacks.
+  auto host = upstream_request.upstreamHost();
+  const bool need_to_send_load_report =
+      (host != nullptr) && (cluster_ != nullptr) && cluster_->lrsReportMetricNames().has_value();
+  if (!need_to_send_load_report) {
+    return;
+  }
+
+  absl::StatusOr<xds::data::orca::v3::OrcaLoadReport> orca_load_report =
```

Comment:
High-level question related to the possible coupling of LRS and ORCA-based LB: IIUC LRS reports are filtered by some value of `LrsReportMetricNames`, but does this imply that the LB will only be effected by the filtered metrics or by all of them?

Reply:
ORCA-based LRS reports will only include metrics configured in `LrsReportMetricNames`. 
It is up to LRS report consumer to make LB decisions on some or all of them.

There will be a separate follow-up PR to implement `ClientSideWeightedRoundRobinLoadBalancer` that will NOT use LRS reports, but rather consume ORCA load reports directly.